### PR TITLE
DG updates

### PR DESCRIPTION
**Pull request by chao890**
DG updates for my part and some extras
- add-leave UML and implementation explanations
- delete-leave UML and implementation explanations
- Update `delete` related diagrams and images

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -30,7 +30,7 @@ Refer to the guide [_Setting up and getting started_](SettingUp.md).
 
 ### Architecture
 
-<img src="images/diagrams/ArchitectureDiagram.png" width="280" />
+<img src="images/diagrams/DeleteCommandUML.png" width="280" />
```
Comment (kevinchangjk) [inline docs/DeveloperGuide.md:33]: Why this change?

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -322,6 +322,64 @@ Step 3. The `execute` method of `ViewCommand` is then called on the object, whic
...
+Step 3. The `execute` method of `AddLeaveCommand` is then called on the object.
+
+Step 4. This calls the `model#getPersonList` method, which returns a list of employees.
+
+Step 5. The target employee is then taken from the list by their employee ID from the input.
+
+Step 6. The `Person#addLeave` method is then called on the employee, and the new leave period is added to the employee.
+
+Step 7. This returns a `CommandResult` object, which is returned to the `LogicManager`.
+
```
Comment (prit3010) [inline docs/DeveloperGuide.md:353]: Design Considerations?

**Review by prit3010 — COMMENTED** [review]
Maybe if got time can add design considerations
